tabular/src/autogluon/tabular/models/tabular_nn/categorical_encoders.py

```diff
@@ -166,7 +166,7 @@ def _check_X(self, X):
         """
         if not (hasattr(X, 'iloc') and getattr(X, 'ndim', 0) == 2):
             # if not a dataframe, do normal check_array validation
-            X_temp = check_array(X, dtype=None)
+            X_temp = check_array(X, dtype=None, force_all_finite=False)
```

Comment:
This line fixed XGB, all other code changes are for FastAI.

@jwmueller This code is pretty complicated so I'm unsure if there are any downsides to this fix.

Reply:
I believe this should be ok